File: day1.c
#include <stdio.h>
#include <stdlib.h>
#include "utils.h"

int main(void) {
    FILE *fp = fopen("inputs/day1.txt", "r");
    // I'll skip the formalities, it's just a coding challenge XD
    char *line = NULL;
    size_t len = 0;
    
    unsigned int aux = 0, max[3] = {0, 0, 0}, i;
    while(getline(&line, &len, fp) != -1) {
        if(line[0] != '\n') {
            aux += atoi(line);
            continue;
        }

        // Pretty sure this can be improved, can't bother
        for(i = 0; i < 3; i++) {
            if(aux > max[i]) {
                max[i] = aux;
                // sort from smallest (pos 0) to largest
                qsort(max, 3, sizeof(unsigned int), compare);
                break;
            }
        }
        aux = 0;
    }

    printf("The top elf is carrying %d calories.\n", max[2]);
    printf("The top three elves are carrying %d calories in total.\n", max[0] + max[1] + max[2]);

    fclose(fp);
    if(line) 
        free(line);
    return 0;
}

File: aoc/day1.py
f = open("inputs/day1.txt", "r")
lines = f.readlines()
f.close()

aux = 0
max_rank = [0, 0, 0]

for line in lines:
    if not len(line.strip()) == 0:
        aux += int(line)
        continue

    # If it's bigger than the first, update the ranks accordingly
    if aux > max_rank[0]:
        max_rank[0] = aux
        max_rank.sort()

    aux = 0

print("The top elf is carrying %d calories." % max_rank[0])
print("The top three elves are carrying %d calories in total." % sum(max_rank))